packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -68,7 +114,11 @@ export class Wallet extends EventEmitter<ObjectiveProposed> {
    * @returns A promise that resolves to a collection of ObjectiveResult.
    */
   public async approveObjectives(objectiveIds: string[]): Promise<ObjectiveResult[]> {
-    const {objectives, messages} = await this._engine.approveObjectives(objectiveIds);
+    const {objectives, messages, chainRequests} = await this._engine.approveObjectives(
+      objectiveIds
+    );
+    const transactions = await this._chainService.handleChainRequests(chainRequests);
+    await Promise.all(transactions.map(tr => tr.wait()));
```

Comment:
Are we sure that we want to wait for transaction mining before resolving this API call?

Reply:
That's a good point! For merging reasons I've updated this upstream in 9496f3a